!!!FEATURE: Reform foundation and boot process of the UI

grebaldi opened the pull request:
refs:  #3119
next: https://github.com/neos/neos-ui/pull/3773

This is the first in a series of PRs that will enable the creation of multiple applications based on a Neos UI core.

This PR turns the `Backend/Index.html` Fluid template into a dedicated `ApplicationView`.  This `ApplicationView` has the sole purpose of building the foundational HTML document (via string concatenation) for bootstrapping the Neos UI.

Formerly, Fusion was used to retrieve and render the initial data for the UI. This task has been delegated to a set of dedicated classes, namely:

- `ConfigurationProvider` - retrieves the `nodeTree` and `structureTree` segments from `Neos.Neos.userInterface.navigateComponent` settings, `allowedTargetWorkspaces` from the ContentRepository's `WorkspaceService` as well as the `nodeTypeSchema` and `translations` endpoints.
- `RoutesProviderInterface` - retrieves all other routes/endpoints required for communication with the Neos server application
- `FrontendConfigurationProvider` - reads and preprocesses the `Neos.Neos.Ui.frontendConfiguration` settings that are mostly used by third-party plugins to share data between server and client
- `NodeTypeGroupsAndRolesProviderInterface` - retrieves information about node type roles and node type groups. Roles are a dedicated UI-concept that is meant to distinguish between document, content and collection nodes. Groups refer the grouping of node types in the creation dialog.
- `MenuProvider` - retrieves all data needed to render the main burger menu located in the top left corner of the UI.
- `InitialStateProvider` - reads and preprocesses the `Neos.Neos.Ui.initialState` settings that are used to hydrate the UI's redux store

The responsibilities of these classes is entirely derived from their former Fusion counterparts. All of them have been marked `@internal` to allow for later removal. Nonetheless, it remains possible to implement their accompanying interfaces and replace their implementations via `Objects.yaml`. This enables rare edge cases and unplanned extensibility.

The splash screen, formerly a Fluid template, is now hard-coded into the `ApplicationView` class, which renders the `Neos.Neos.Ui.splashScreen` setting obsolete. It has therefore been removed.

Furthermore, this PR reforms the client-side boot process of the UI. This includes the following changes:

* Scripts (The UI main script, as well as plugins) are now loaded in the `<head/>` with `defer`
  * This way, the UI does not need to listen to DOMContentLoaded in order to discover the #appContainer element. This allows a lot of the UI boot process to become synchronous.
* The main boot function is no longer a Redux Saga. It is instead an asynchronous function.
  * Also, the various boot-related sub-processes (initializing plugins, loading translations, etc...) have been wrapped in dedicated functions.
* The initial data for the boot process is now `JSON.parse`d instead of `window['_NEOS_UI_*']`ed
  * `JSON.parse` is somewhat more performant, because for JSON only a subset of ECMAScript needs to be parsed. (Though, this is not all that significant given the amount of data, but still :slightly_smiling_face:)
* The `data-first-tab` attribute has been removed from #appContainer
  * I was unable to find any references to it within the code base, so I removed it, because I didn't know what exactly to put in there. I took it for a historical leftover. Please let me know if I'm mistaken :)

**Remaining TODOs**

- [x] Replace Fusion with a dedicated application view class
- [x] Use `<head>`-scripts + `defer` instead of `<body>`-scripts for script inclusion
- [-] <s>Turn the remaining Fluid templates into Fusion prototypes as well (optional)</s>
  - [-] <s>`Backend/ConfigurationVersion.html`</s>
  - [-] <s>`Backend/Guest.html`</s>
  - [-] <s>`Backend/GuestNotificationScript.html`</s>
  - [-] <s>`Error/ErrorMessage.html`</s>
  - **EDIT:** Scope is large enough as it is - these can be done later
- [x] Remove the `actions.System.boot()` dispatch?
  - Noone seems to listen to it...
- [x] Remove `Neos.Neos.Ui.splashScreen` configuration
  - Relies on fluid partial that no longer exists
  - Replacement? -> **EDIT:** None. It's hard-coded now. We can open this up later if there's demand for it.
- [-] <s>Remove fusion objects that are no longer relevant</s>
  - [-] <s>`Neos.Neos.Ui:RenderConfiguration`</s>
  - **EDIT:** Nope, this is still being used for the guest frame
- [-] <s>Deprecate EEL-Helpers that are no longer relevant</s>
  - [-] <s>`Neos.Ui.Workspace`</s>
  - **EDIT:** Nope, this is still being used in settings
- [x] Add doc comment that explains node type roles and node type groups concepts
- [x] Rename `NodeTypesProviderInterface` -> `NodeTypeGroupsAndRolesProviderInterface`
- [x] Use an interface to className mapping in `Objects.yaml` instead of explicitly configuring `BackendController` properties
- [x] Prevent duplication of `defer` attribute on `<script/>`-includes through `Neos.Neos.Ui.resources` settings

**Note to fellow maintainers**

In order to preserve the usefulness of the commits in this PR, please do not merge any of the version branches into this branch. To update the code with the latest changes in `9.0`, please do an interactive rebase instead:

```
git fetch upstream
git checkout -b feature/multi-app/01-foundation upstream/feature/multi-app/01-foundation
git rebase -i upstream/9.0 feature/multi-app/01-foundation
```

This will require
```
git push -f upstream feature/multi-app/01-foundation
```
afterwards.

This way, every commit will be replayed on top of `9.0`, and can be adjusted seperately.

daniellienert committed:
TASK: Refactor controller to not use static fusion path

The fusion path is now dynamically determined by controller / action
This enables additional actions with separate templates for that controller.

grebaldi committed:
TASK: Load head scripts using `<script defer/>`

This way, the UI does not need to listen to `DOMContentLoaded` in order
to discover the `#appContainer` element. This allows a lot of the UI
boot process to become synchronous.

grebaldi committed:
TASK: Refactor the UI boot process

Functionally, it stays the same, but the boot process is no longer a
Redux Saga. It is instead an asynchronous function that runs when the
`DOMContentLoaded` event was fired.

Also, the various boot-related sub-processes (initializing plugins,
loading translations, etc...) have been wrapped in dedicated functions.

grebaldi committed:
TASK: Remove `discover` function from utility-helpers package

As it is no longer needed.

grebaldi committed:
TASK: Remove `delay` function from utility-helpers package

As it is no longer needed.

grebaldi committed:
TASK: JSON.parse the initial data for the boot process...

...rather than inlineing it. `JSON.parse` is somewhat more efficient,
because for JSON only a subset of ECMAScript has to be parsed.

grebaldi committed:
TASK: Refactor fatal initlization error handling

The error message has been adjusted to the look of the global
ErrorBoundary, re-using its stylesheet.

grebaldi committed:
TASK: Group all initial data variables into a single view variable

grebaldi committed:
TASK: Turn SplashScreen into a Fusion Component

grebaldi committed:
TASK: Remove `data-first-tab` attribute from #appContainer

This attributes has no references throughout the entire code base. It
appears to be a leftover from some abandoned feature-work.

grebaldi committed:
TASK: Move fusion declaration of initial data into a separate prototype

grebaldi committed:
TASK: Move fusion declaration of head stylesheets into a separate prototype

grebaldi committed:
TASK: Move fusion declaration of head icons into a separate prototype

grebaldi committed:
TASK: Move fusion declaration of headScripts into a separate prototype

grebaldi committed:
FEATURE: Introduce `Neos.Neos.Ui:Application` prototype

The `Neos.Neos.Ui:Application` prototype is designed to:

1. Functionally replace the `Backend/Index.html` Fluid template
2. Act as a basis for future multi-application support

As of this commit, the API of the `Neos.Neos.Ui:Application` is not
final.

grebaldi committed:
TASK: Use the `Neos.Neos.Ui:Application` prototype to render the UI

grebaldi committed:
TASK: Remove the `Backend/Index.html` Fluid template

...as it is no longer needed.

grebaldi committed:
TASK: Make fusion includes local to the prototypes that use them

mhsdesign commented:
Thank you so much!!! This really cleans up many things that are needlessly messy and hard to understand. I added already a dec comments to your text and will look over the code later ;)

> Remove the actions.System.boot() dispatch?

I only know that I have used the init see:
https://github.com/neos/neos-ui/issues/3098#issuecomment-1303492943
(And just noticed the issue might be generally interesting for you;))

About:
> Remove Neos.Neos.Ui.splashScreen configuration

and
> The data-first-tab attribute has been removed from #appContainer

I have never heard of those things so I assume they can go;)


> are now loaded in <body/> instead of <head/>

Today I actually prefer the use of the defer attribute … feels cleaner to have them in the head … but it’s really not important 

--------

Note to myself:
the commit is probably the one which will get outdated the easiest and has to be taken care of carefully.
> TASK: Group all initial data variables into a single view variable

Also i think we should incorporate https://github.com/neos/neos-ui/pull/2920 into here.

-------

Thanks again!!!! :D

mhsdesign commented on Resources/Private/Fusion/Backend/Component/HeadIcons.fusion:
```diff
@@ -0,0 +1,57 @@
+ * source code.
+ */
+
+include: resource://Neos.Fusion/Private/Fusion/Root.fusion
```
this is sadly not a good idea to use fusion imo. The fusion parser will / cannot optimize this and the whole neos.fusion stuff is reparsed. That has to be done unfortunately because fusion allows global paths that can be overridden.

grebaldi replied:
You're confirming my worst fears there :sweat_smile:

mhsdesign commented on Resources/Private/Fusion/Backend/Component/Scripts.fusion:
```diff
@@ -0,0 +1,16 @@
+
+/** @internal */
+prototype(Neos.Neos.Ui:Component.Scripts) < prototype(Neos.Fusion:Value) {
+    value = ${scripts}
```
i know this is just a draft, but i wanted to point out that this context variable `scripts` is super sneaky and unexpected here ^^

grebaldi commented:
Hi @mhsdesign,

thanks a lot for your very valuable feedback :+1: 

> Today I actually prefer the use of the defer attribute … feels cleaner to have them in the head … but it’s really not important

I'm always afraid of introducing race conditions when it comes to this stuff :sweat_smile:. Well, MDN states:

> This Boolean attribute is set to indicate to a browser that the script is meant to be executed after the document has been parsed, but before firing [DOMContentLoaded](https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event).
(see: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#defer)

So, I guess this means we have a guarantee that the script can see the whole document? I mean `document has been parsed` sounds like that, but `before firing DOMContentLoaded` strikes me the wrong way - so as if we're at a point of execution at which the state of the DOM isn't exactly determined.

I'm with you, that we should use `defer`, especially if that means that the browser's rendering thread doesn't have to block for parsing our UI bundle before displaying anything. So, if you can take my above stated fear from me, I'll gladly change the script integration to `defer` :D

I also have some ideas bouncing around in my head with regards to the code issues you've pointed out. I similarly feel that the fusion foundation as of right now is less than ideal. I see two possible way out of here:

**A) Build dedicated Fusion `@class` implementations for resource collection** 

So, basically, this would mean to back the following prototypes with classes:
- `Neos.Neos.Ui:Component.InitialData`
- `Neos.Neos.Ui:Component.HeadStylesheets`
- `Neos.Neos.Ui:Component.HeadIcons`
- `Neos.Neos.Ui:Component.Scripts`

This would move all the resource-collecting logic out of the controller and out of the fusion code. It'd also eliminate the need for `include: resource://Neos.Fusion/Private/Fusion/Root.fusion` (at least I think it would :)).

**B) Replace Fusion with a dedicated application view class**

Because: what's the point of using Fusion to initialize the UI anyway?

A dedicated view could take care of hiding the resource-collecting logic from the controller. The `initialData` contents would map nicely onto view variables. The view could just concatenate the HTML document - nice, short and fast :slightly_smiling_face: 

wdyt?

grebaldi commented:
Oh, and regarding #3098:

I don't think that this issue would be fixed as of right now. But I'm going to try to move around some bits of the boot process (should be possible), so plugins have an easier time reading this stuff.

mhsdesign commented:
I actually dont know that much about `defer` - just used it - so lets stick at first with js body scripts ^^

> B) Replace Fusion with a dedicated application view class
> Because: what's the point of using Fusion to initialize the UI anyway?

And yeah i thought so too... well lets just use b? :D
The only thing would be of course unplanned extensibility, but i havent used that and everything useful is already available as api via settings. 

especially things like
```
# TODO: dirty hack to not have to re-implement neos:backend.configurationCacheVersion VH
version = Neos.Fusion:Template {
    templatePath = 'resource://Neos.Neos.Ui/Private/Templates/Backend/ConfigurationVersion.html'
    @process.trim = ${String.trim(value)}
}
```
make me rethink if fusion is the right choice 😅 

Aaand we might finally remove this weirdly namespaced `Neos.Ui.PositionalArraySorter.sort` and `Neos.Ui.Api` helpers 😂

grebaldi commented:
@mhsdesign 

> I actually dont know that much about `defer` - just used it - so lets stick at first with js body scripts

I looked into the spec, but it is just as imprecise about the exact timing as MDN is:
> If the [async](https://html.spec.whatwg.org/multipage/scripting.html#attr-script-async) attribute is not present but the [defer](https://html.spec.whatwg.org/multipage/scripting.html#attr-script-defer) attribute is present, then the classic script will be fetched [in parallel](https://html.spec.whatwg.org/multipage/infrastructure.html#in-parallel) and evaluated when the page has finished parsing.
>
(see: https://html.spec.whatwg.org/multipage/scripting.html#attr-script-defer)

Then I found this article that is very confident in saying:
> Whereas DEFER scripts don’t execute until the HTML document is done being parsed (AKA, DOM Interactive or `performance.timing.domInteractive`).
>
(see: https://calendar.perfplanet.com/2016/prefer-defer-over-async/#attributes)

Sadly no sources to back that up...

I guess it wouldn't make sense for any `defer` implementation to not have the DOM available after document parsing. So, I think we can relatively safely assume that `<head>`-scripts + `defer` will work fine. I'll add that to the list :)

> The only thing would be of course unplanned extensibility

Actually, you'd have to do quite some acrobatics to hook into the backend fusion view. It cannot be extended via regular ol' `autoInclude` - which was by design, because this view is not supposed to be interfered with.

I think the reasoning for using fusion back then was that it provided a unified high-level API for both MVC/Routing and ContentRepository, which are both major concerns for initial hydration of the UI. But this was always supposed to be a strictly hidden implementation detail.

> [...] everything useful is already available as api via settings.

https://github.com/neos/neos-ui/blob/2204bb11070d8fc43bb28cdbc1d208cefa255b91/Configuration/Settings.yaml#L94-L96

:point_up: Well... Actually those settings are not supposed to be API. However, this is certainly not the best way to hide stuff :sweat_smile: 

I'd like to get rid of this mechanism as well, but that exceeds the scope of this PR.

> well lets just use b? 

I'll add that to the list as well :)

mhsdesign commented:
Thanks for the housekeeping and implementing the `defer` :D

> Actually, you'd have to do quite some acrobatics to hook into the backend fusion view. It cannot be extended via regular ol' autoInclude - which was by design, because this view is not supposed to be interfered with.

yes i fully agree and also talked to @Sebobo about this. There is absolutely no need to have it in fusion and php would be a better fit.

But to keep this pr rather simpler, i would say that we could keep the fusion code in one messy file for now and refactor it to php with another followup pr?

I especially dont want to merge the `include: resource://Neos.Fusion/Private/Fusion/Root.fusion` part :D

But if you think its doable in scope of this pr i wouldnt mind moving to php quicker.

grebaldi commented:
> But if you think its doable in scope of this pr i wouldnt mind moving to php quicker.

Already on it :D

grebaldi committed:
!!!FEATURE: Introduce `ApplicationView` to replace Fusion for bootstrapping the UI

The `ApplicationView` builds the foundational HTML document (via string
concatenation) for bootstrapping the Neos UI.

Formerly, Fusion was used to retrieve and render the initial data for
the UI. This task has been delegated to a set of dedicated classes,
namely:

- `ConfigurationProvider` - retrieves the `nodeTree` and `structureTree`
segments from `Neos.Neos.userInterface.navigateComponent` settings,
`allowedTargetWorkspaces` from the ContentRepository's
`WorkspaceService` as well as the `nodeTypeSchema` and `translations`
endpoints.
- `RoutesProviderInterface` - retrieves all other routes/endpoints
required for communication with the Neos server application
- `FrontendConfigurationProvider` - reads and preprocesses the
`Neos.Neos.Ui.frontendConfiguration` settings that are mostly used by
third-party plugins to share data between server and client
- `NodeTypeGroupsAndRolesProviderInterface` - retrieves information about
node type roles and node type groups. Roles are a dedicated UI-concept that
is meant to distinguish between document, content and collection nodes.
Groups refer the grouping of node types in the creation dialog.
- `MenuProvider` - retrieves all data needed to render the main burger
menu located in the top left corner of the UI.
- `InitialStateProvider` - reads and preprocesses the
`Neos.Neos.Ui.initialState` settings that are used to hydrate
the UI's redux store

The responsibilities of these classes is entirely derived from their
former Fusion counterparts. All of them have been marked `@internal` to
allow for later removal. Nonetheless, it remains possible to implement
their accompanying interfaces and replace their implementations via
`Objects.yaml`. This enables rare edge cases and unplanned
extensibility.

The splash screen, formerly a Fluid template, is now hard-coded into the
`ApplicationView` class, which renders the `Neos.Neos.Ui.splashScreen`
setting obsolete. It has therefore been removed.

grebaldi committed:
TASK: Remove all backend-related Fusion code

With the new `ApplicationView` this code is no longer in use anywhere.

mhsdesign commented on Classes/Domain/NodeTypesProviderInterface.php:
```diff
@@ -0,0 +1,26 @@
+namespace Neos\Neos\Ui\Domain;
+
+/**
+ * @internal
```
your commit message expresses really well the definition of the roles and id very much like to have that as a doc comment here ^^

> Retrieves information about node type roles and
node type groups.
>
> Roles are a dedicated UI-concept that is meant to
distinguish between document, content and collection nodes. Groups refer
the grouping of node types in the creation dialog.

Additionally i would like to propose a different more obvious and specific naming like `NodeTypeGroupsAndRolesProviderInterface`

mhsdesign commented on Configuration/Objects.yaml:
```diff
@@ -0,0 +1,24 @@
+Neos\Neos\Ui\Controller\BackendController:
+  properties:
+    configurationProvider:
+      object: Neos\Neos\Ui\Infrastructure\Configuration\ConfigurationProvider
+    routesProvider:
+      object: Neos\Neos\Ui\Infrastructure\MVC\RoutesProvider
+    frontendConfigurationProvider:
+      object: Neos\Neos\Ui\Infrastructure\Configuration\FrontendConfigurationProvider
+    nodeTypesProvider:
+      object: Neos\Neos\Ui\Infrastructure\ContentRepository\NodeTypesProvider
+    menuProvider:
+      object: Neos\Neos\Ui\Infrastructure\Neos\MenuProvider
+    initialStateProvider:
+      object: Neos\Neos\Ui\Infrastructure\Configuration\InitialStateProvider
```
I would propose to go for a interface to className mapping instead of declaring just the properties for the backend-controller. Because the ConfigurationProvider also uses the `CacheConfigurationVersionProviderInterface` and via a global interface override it would become obvious how the dependencies are resolved.


```yaml
'Neos\Neos\Ui\Domain\NodeTypesProviderInterface':
  className: Neos\Neos\Ui\Infrastructure\ContentRepository\NodeTypesProvider
# ...
```

mhsdesign reviewed (commented):
WOW ❤️ 🎉 thank you so much for so cleanly moving the neos ui integration to php.
The code looks way more obvious and is easier to follow! Thanks a lot!

mhsdesign commented on Classes/Domain/Service/StyleAndJavascriptInclusionService.php:
```diff
@@ -61,7 +61,7 @@ class StyleAndJavascriptInclusionService
     {
         return $this->build($this->javascriptResources, function ($uri, $additionalAttributes) {
-            return '<script src="' . $uri . '" ' . $additionalAttributes . '></script>';
+            return '<script src="' . $uri . '" ' . $additionalAttributes . ' defer></script>';
```
if one configures 

```yaml
resource: fooo
defer: true
```

or
```yaml
resource: fooo
attributes:
  defer: true
```

there will be actually a second `defer` attribute in $additionalAttributes already 🤯 shouldn't be a real problem though ^^

i could imagine omitting the `defer` below in the `htmlAttributesArrayToString`.

And it seems that if someone uses an esm plugin with 

```yaml
resource: fooo
attributes:
  type: "module"
```

we can omit this defer tag, as its implicitly set and doppeltgemoppelt? https://wiki.selfhtml.org/wiki/HTML/Attribute/defer

mhsdesign replied:
But i could confirm that `<script src="plugin" type="module" defer=""></script>` works ;)

mhsdesign replied:
And i see you fixed my initial concerns via https://github.com/neos/neos-ui/pull/3682/commits/4325fb0d5c239954e37491cf56c683ad4a85dbfd ;)

mhsdesign commented:
Fyi this will be a bit nasty to rebase to https://github.com/neos/neos-ui/pull/3689 but i think doable, so i would propose to merge that first?

grebaldi committed:
TASK: Remove `@neos/neos-ui/System/BOOT` redux action + dispatch

This action used to be fired before all asynchronous operations at the
beginning of the boot process.

Since there's not a single subscriber within the UI code base that
listens to that action, it has now been removed.

grebaldi commented:
@mhsdesign 
> Fyi this will be a bit nasty to rebase to https://github.com/neos/neos-ui/pull/3689 but i think doable, so i would propose to merge that first?

#3689 should definitely go first. I don't think rebase will be that big a deal :)

grebaldi committed:
TASK: Prevent duplication of `defer` attribute on `<script/>`-includes

This also removes handling of the legacy first level 'defer' setting in
the `Neos.Neos.Ui.resources` settings. If it is set, it will be ignored.

mhsdesign commented:
Id say this is nearly ready to be merged. Because im a little paranoid i actually diffed the initial bootstrap html before and after and found a few differences worth noting.

First it seems that the `User Settings` module link was not included previously, but now shows up and will be rendered in the menu:

```json
        {
            "label": "Neos.Neos:Modules:user.label",
            "icon": "fas fa-users",
            "children": [
                {
                    "icon": "fas fa-user",
                    "label": "Neos.Neos:Modules:userSettings.label",
                    "uri": "http:\/\/localhost:8081\/neos\/user\/usersettings",
                }
            ]
        }
```

There seems to be a `module.hideInMenu` or `submodule.hideInMenu` flag to toggle this.

<img width="386" alt="image" src="https://github.com/neos/neos-ui/assets/85400359/a57bf6d0-8fd4-4ba8-b752-aeea16fdbbf8">

And second, it seems that we boot the ui with more data: I found that everything mapped to the legacy globals but there is a new entry `settings` with seems to be an complete dump of the whole Neos.Neos.Ui configuration, at least by looking at its keys:

- 'resources'
- 'contentCanvas'
- 'frontendConfiguration'
- 'frontendDevelopmentMode'
- 'nodeTypeRoles'
- 'configurationDefaultEelContext'
- 'documentNodeInformation'
- 'initialState'
- 'changes'
- 'outOfBandRendering'

Edit. I found the cause. We json_encode all the $this->variables, but it was once thought be a good feature to have the settings automagically available see https://github.com/neos/flow-development-collection/commit/b944521345d1801ce146a3aa6be80b5871b3224e.
So i guess we need to namespace all the variables in something like `booting` or `value` like the json view idk.

mhsdesign committed:
TASK: `terminateDueToFatalInitializationError` throw directly if no dom is initialized yet

mhsdesign committed:
BUGFIX: Avoid rendering unwanted variables in the view by scoping everything into `initialData`

Previously this feature https://github.com/neos/flow-development-collection/commit/b944521345d1801ce146a3aa6be80b5871b3224e would cause also the rendering of the neos ui `settings`

mhsdesign committed:
BUGFIX: Hide first level menus if `hideInMenu` set

mhsdesign reviewed (approved):
I adjusted the previously mentioned parts a little and now everything is like expected.
I would prefer another or two reviews though ^^
Thanks again for the hard work!

mhsdesign commented on Classes/Domain/CacheConfigurationVersionProviderInterface.php:
```diff
@@ -0,0 +1,27 @@
+
+declare(strict_types=1);
+
+namespace Neos\Neos\Ui\Domain;
```
shouldnt we group all these `ProviderInterface`s into one sub namespace like `Neos\Neos\Ui\Domain\BackendProviders`?

That way `Neos\Neos\Ui\Domain` is not that cluttered and there is also "space" for other interfaces not directly related to the backend, like the NodeCreationHandlerInterface for example?

I could take care of that refactoring, but i wanted to ask for your blessing @grebaldi ^^

mhsdesign replied:
Or maybe `Neos\Neos\Ui\Domain\InitialData` or `Neos\Neos\Ui\Domain\Booting` or `Neos\Neos\Ui\Domain\InitialDataProvider`?

mhsdesign replied:
Discussed this with Wilhelm and he was okay with moving it to `InitialData`.

mhsdesign committed:
TASK: Move Providers to `InitialData` namespace

Sebobo commented on Configuration/Views.yaml:
```diff
@@ -1,7 +1,6 @@
 -
   requestFilter: 'isPackage("Neos.Neos.Ui") && isController("Backend")'
-  viewObjectName: 'Neos\Fusion\View\FusionView'
+  viewObjectName: 'Neos\Neos\Ui\Presentation\ApplicationView'
```
Do we still need the Views.yaml as the ApplicationView is already assigned in the Controller?

mhsdesign replied:
yes we should probably remove that.

mhsdesign replied:
Fixed via https://github.com/neos/neos-ui/pull/3682/commits/ba8cf91b2df323fd112945229362173f5ccadd46 now using the $defaultViewObjectName as that is more predictable and views yaml is rather considered for extensibility nowadays

Sebobo reviewed (approved):
LGTM, just left one question regarding the necessity of the Views.yaml

mhsdesign committed:
TASK: Remove Views.yaml in favour of $defaultViewObjectName
